base_report_to_printer/report.py

```diff
@@ -25,12 +25,26 @@
 class Report(models.Model):
     _inherit = 'report'
 
+    def _can_send_report(self, cr, uid, ids, behaviour, printer, document,
+                         context=None):
+        """Predicate that decide if report can be sent to printer
+
+        If you want to prevent `get_pdf` to send report you can set
+        the `must_skip_sent_to_printer` key to True in the context
+        """
+        if context is None:
+            context = self.pool['res.users'].context_get(cr, uid)
+        if context.get('must_skip_sent_to_printer'):
+            return False
+        if behaviour['action'] == 'server' and printer and document:
+            return True
+        return False
+
     def print_document(self, cr, uid, ids, report_name, html=None,
                        data=None, context=None):
         """ Print a document, do not return the document file """
-        document = super(Report, self).get_pdf(cr, uid, ids, report_name,
-                                               html=html, data=data,
-                                               context=context)
+        document = self.get_pdf(cr, uid, ids, report_name,
+                                html=html, data=data, context=context)
```

Comment:
`get_pdf` should be called with the must_skip_sent_to_printer key to avoid to be printed 2 times